firebase-firestore-mixin.html

```diff
@@ -169,10 +169,11 @@
 
         this._firestoreProps = {};
         this._firestoreListeners = {};
-        this.db = this.constructor.db || firebase.firestore();
       }
 
       connectedCallback() {
+        super.connectedCallback();
+        this.db = this.constructor.db || firebase.firestore();
```

Comment:
Do we ever have `this.constructor.db` defined? I think we could improve it even further:
- actually lazy-require Firestore (if binding is called with all props defined, line 231?)
- introduce a helper method for this
- we could move it out of the mixin entirely (to the scope), unless we want to give an access to `db` prop, then we would need to make some docs for it

Reply:
> Do we ever have this.constructor.db defined?

Doesn't seem so...copy pasta.

> actually lazy-require Firestore (if binding is called with all props defined, line 231?)

Yes a `this.db = this.db || firebase.firestore();` there would seem sufficient.

> we could move it out of the mixin entirely (to the scope), unless we want to give an access to db prop, then we would need to make some docs for it

I'm not quite sure I follow.